Data/ByteString.hs

```diff
@@ -1527,13 +1527,25 @@ isInfixOf p s = null p || not (null $ snd $ breakSubstring p s)
 --
 -- @since 0.11.2.0
 isValidUtf8 :: ByteString -> Bool
-isValidUtf8 (BS ptr len) = accursedUnutterablePerformIO $ unsafeWithForeignPtr ptr $ \p -> do 
-  i <- cIsValidUtf8 p (fromIntegral len)
+isValidUtf8 (BS ptr len) = accursedUnutterablePerformIO $ unsafeWithForeignPtr ptr $ \p -> do
+  -- Use a safe FFI call for large inputs to avoid GC synchronization pauses
+  -- in multithreaded contexts.
+  i <- if len < 1000000
```

Comment:
It would be good to have a comment that explains why the cutoff was chosen to be 1000000 bytes. If it's just a random guess, just say that it's a random guess.

Reply:
Would it be enough to link to my comment here: https://github.com/haskell/bytestring/issues/451#issuecomment-991879338

Reply:
Good enough for me. :)